onlinejudge/service/yukicoder.py

```diff
@@ -346,17 +346,14 @@ def from_url(cls, url: str) -> Optional['YukicoderProblem']:
         dirname, basename = posixpath.split(utils.normpath(result.path))
         if result.scheme in ('', 'http', 'https') \
                 and result.netloc == 'yukicoder.me':
-            n = None  # type: Optional[int]
             try:
                 n = int(basename)
             except ValueError:
-                pass
-            if n is not None:
-                if dirname == '/problems/no':
-                    return cls(problem_no=int(n))
-                if dirname == '/problems':
-                    return cls(problem_id=int(n))
-            return cls()
+                return None
+            if dirname == '/problems/no':
+                return cls(problem_no=int(n))
```

Comment:
```suggestion
                return cls(problem_no=n)
```

`n` is already an `int`